### PR TITLE
Add option to save copy of database at regular intervals

### PR DESCRIPTION
**Pull request by joshuacolvin0**

### CODE REVIEW

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 1]
```diff
@@ -788,6 +799,30 @@ void ArbCore::operator()() {
...
+                    auto current_seconds = seconds_since_epoch();
+                    if (next_rocksdb_save_timestamp != 0 &&
+                        current_seconds >= next_rocksdb_save_timestamp) {
+                        auto timestamp_dir = std::to_string(current_seconds);
+                        auto checkpoint_dir = save_rocksdb_path / timestamp_dir;
+                        status =
+                            tx.createRocksdbCheckpoint(checkpoint_dir.string());
+                        if (!status.ok()) {
+                            std::cerr << "Unable to save checkpoint into "
+                                      << checkpoint_dir << std::endl;
```
Comment (PlasmaPower) [inline packages/arb-avm-cpp/data_storage/src/arbcore.cpp:813]: I'd print `status.ToString()` here

**Review by PlasmaPower — APPROVED** [review]
LGTM, just one comment on error logging

**Comment by PlasmaPower** [comment]
Looks like there's a minor issue with the tests, CI is failing: https://app.circleci.com/pipelines/github/OffchainLabs/arbitrum/9495/workflows/f048e77e-818e-481b-98de-02682d64e80d/jobs/52599

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp (hunk 1 above)
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/src/arbcore.cpp]: :+1:

**Review by PlasmaPower — DISMISSED** [review]
LGTM

**Review by PlasmaPower — APPROVED** [review]
LGTM

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**